File: Chat/Server/src/Chat/Script/CsScript/Action/Action3001.cs
using GameServer.Script.CsScript.Com;
using GameServer.Script.Model;
using ZyGames.Framework.Game.Contract;
using ZyGames.Framework.Game.Service;
using ZyGames.Framework.RPC.Sockets;

namespace GameServer.Script.CsScript.Action
{
    /// <summary>
    /// Send chat
    /// </summary>
    public class Action3001 : BaseAction
    {
        private string _message;


        public Action3001(ActionGetter actionGetter)
            : base((short)3001, actionGetter)
        {
            IsNotRespond = true;
        }

        /// <summary>
        /// 客户端请求的参数较验
        /// </summary>
        /// <returns>false:中断后面的方式执行并返回Error</returns>
        public override bool GetUrlElement()
        {
            if (httpGet.GetString("message", ref _message) && !string.IsNullOrEmpty(_message))
            {
                return true;
            }
            return false;
        }

        /// <summary>
        /// 业务逻辑处理
        /// </summary>
        /// <returns>false:中断后面的方式执行并返回Error</returns>
        public override bool TakeAction()
        {
            var chatservice = new MyChatService(Current.User as ChatUser);
            chatservice.Send(_message);
            return true;
        }

        protected override string BuildResponsePack()
        {
            Body = true;
            return base.BuildResponsePack();
        }


        public override void TakeActionAffter(bool state)
        {
            ActionFactory.SendAsyncAction(GameSession.GetOnlineAll(), 3002, null, (s, r) => { }, OpCode.Text, 0).Wait();
            base.TakeActionAffter(state);
        }
    }
}

File: Chat/Server/src/Chat/Script/PyScript/Action/action3001.py
"""Send chat"""
import clr, sys
import json
import ReferenceLib
from action import *
from ZyGames.Framework.Common import *
from ZyGames.Framework.Game.Contract import *
from ZyGames.Framework.Game.Service import *
from ZyGames.Framework.RPC.Sockets import *
from GameServer.Script.CsScript.Com import *
from GameServer.Script.Model import *
#引用示例
#clr.AddReference('ZyGames.Framework.Common');
#from ZyGames.Framework.Common.Log import *

class UrlParam(HttpParam):
    def __init__(self):
        HttpParam.__init__(self)


class ActionResult(DataResult):
    def __init__(self):
        DataResult.__init__(self)


def getUrlElement(httpGet, parent):
    urlParam = UrlParam()
    urlParam.MsgId = httpGet.GetMsgId()
    urlParam.ActionId = httpGet.GetActionId()
    if httpGet.Contains("message"):
        urlParam._message = httpGet.GetStringValue("message")
    else:
        urlParam.Result = False

    return urlParam

def takeAction(urlParam, parent):
    actionResult = ActionResult()
    user = parent.Current.User
    chatservice = MyChatService(user)
    chatservice.Send(urlParam._message)
    return actionResult

def buildPacket(urlParam, actionResult):
	
    ActionFactory.SendAction(GameSession.GetOnlineAll(), 3002, None, sendCallback, OpCode.Text, 0);
	
    result = JsonDataResult(urlParam)
    result.setBody(True)
    jsonStr = json.dumps(result, default=convert_to_builtin_type)
    return jsonStr

def sendCallback(session, asyncResult):
    # send to client result.
    return

def convert_to_builtin_type(obj): 
    d = {  }
    d.update(obj.__dict__)
    return d